xwiki-commons-core/xwiki-commons-store/xwiki-commons-store-blob/xwiki-commons-store-blob-filesystem/src/main/java/org/xwiki/store/blob/internal/FileSystemBlobStoreManager.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.store.blob.internal;
+
+import java.nio.file.Path;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.environment.Environment;
+import org.xwiki.store.blob.BlobStore;
+import org.xwiki.store.blob.BlobStoreException;
+import org.xwiki.store.blob.BlobStoreManager;
+
+/**
+ * Blob store manager for the file-system-based blob store.
+ *
+ * @version $Id$
+ * @since 17.7.0RC1
+ */
+@Component
+@Singleton
+@Named("filesystem")
+public class FileSystemBlobStoreManager implements BlobStoreManager
+{
+    @Inject
+    private Environment environment;
+
+    @Override
+    public BlobStore getBlobStore(String name) throws BlobStoreException
+    {
+        Path basePath = this.environment.getPermanentDirectory().toPath().resolve(name);
```

Comment:
Hmm, I feel safer to put those stores in a sub folder, like `store/` (which already exist, but does not currently contain any `filesystem` store, the current file store is in `file`).

We could still have stuff in the permanent directory which we don't want to share with other cluster member but which are still not something that would go in the temporary directory.

Reply:
Very good point, indeed. Keeping the exact same files and just change the API to access them would definitely make upgrades a lot easier.